conda_build/build.py

```diff
@@ -1276,7 +1276,8 @@ def test(recipedir_or_package_or_metadata, config, move_broken=True):
 
     warn_on_use_of_SRC_DIR(metadata)
 
-    config.compute_build_id(metadata.name())
+    if config.set_build_id:
```

Comment:
Actually my example has no tests at all, so it seems the work dir is really not cleaned up when starting a new build with --no-build-id.